jmh-core/src/main/java/org/openjdk/jmh/profile/XCTraceNormProfiler.java

```diff
@@ -0,0 +1,328 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jmh.profile;
+
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.results.*;
+import org.openjdk.jmh.util.FileUtils;
+import org.openjdk.jmh.util.TempFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * macOS perfnorm profiler based on xctrace utility shipped with Xcode Instruments.
+ * Depending on the chosen tracing configuration, so-called template, xctrace may sample PMU counters
+ * using either timer interrupts, or PMU interrupts.
+ * <p>
+ * The profiling process consists of several steps:
+ * 1) launching a program that needs to be profiled using `xctrace record` command; in case of success,
+ * the output of this step is a "trace-file", which is in fact a directory containing multiple files
+ * representing the recorded trace, the trace may contain multiple resulting tables, depending on the template;
+ * 2) inspecting a recorded trace to check if it contains a table supported by the profiler; this information
+ * could be obtained from the trace's table of contents (`xctrace export --toc`);
+ * 3) extracting the table with profiling results from the trace file using `xctrace export` and parsing it
+ * to extract individual samples.
+ * <p>
+ * `xctrace export` extracts data only in XML format, thus both the table of contents and the table
+ * with profiling results need to be parsed as XML documents.
+ * <p>
+ * This profiler currently supports only one table type: counters-profile.
+ * Such tables are generated by the CPU Counters instrument performing sampling either by
+ * timer interrupts, or interrupts generated by a PMU counter, depending on particular configuration.
+ * <p>
+ * Tracing configuration, or template, mentioned earlier is a file that could be configured and saved
+ * using Instruments application. Unfortunately, there is no other way to set it up.
+ * <p>
+ * There's no builtin template that could be used with this profiler, thus the template had to be configured and saved
```

Comment:
Well, there's a way to make it work almost like `perf stat`, but it involves building a temporary Instruments package, so I didn't dare to post the solution (it lives in a separate branch: I was not sure about the copyright header, so plunked the Oracle's one.) .